packages/core/src/server/watchFiles.ts

```diff
@@ -0,0 +1,58 @@
+import type { DevConfig, WatchFiles } from '@rsbuild/shared';
+import type { RsbuildDevMiddlewareOptions } from './getDevMiddlewares';
+
+export async function setupWatchFiles(
+  dev: DevConfig,
+  compileMiddlewareAPI: RsbuildDevMiddlewareOptions['compileMiddlewareAPI'],
+) {
+  const { watchFiles } = dev;
+  if (!watchFiles) {
+    return;
+  }
+
+  const watchFilesOptions = normalizeWatchFilesOptions(watchFiles);
+  if (!watchFilesOptions) {
+    return;
+  }
+
+  const { paths, options } = watchFilesOptions;
+
+  const chokidar = await import('@rsbuild/shared/chokidar');
+  const watcher = chokidar.watch(paths, options);
+
+  if (dev.hmr || dev.liveReload) {
```

Comment:
We can only initialize the Watcher when necessary.

```suggestion
  if (dev.hmr || dev.liveReload) {
    const watcher = chokidar.watch(paths, options);
```

Reply:
> We can only initialize the Watcher when necessary.

ok, this is good